recipes/extra-cmake-modules/all/conanfile.py

```diff
@@ -8,49 +12,65 @@ class ExtracmakemodulesConan(ConanFile):
     description = "KDE's CMake modules"
-    generators = "cmake"
     no_copy_source = False
+    package_type = "build-scripts"
+    settings = "build_type"
+    short_paths = True
 
-    _cmake = None
+    def source(self):
+        get(
+            self,
+            **self.conan_data["sources"][self.version],
+            destination=self.source_folder
+        )
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def build_requirements(self):
+        if self.version <= Version("5.80.0"):
+            self.tool_requires("cmake/[>=3.5]")
+        else:
+            self.tool_requires("cmake/[>=3.16]")
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("extra-cmake-modules-{}".format(self.version), self._source_subfolder)
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_HTML_DOCS"] = False
+        tc.cache_variables["BUILD_QTHELP_DOCS"] = False
+        tc.cache_variables["BUILD_MAN_DOCS"] = False
+        tc.cache_variables["BUILD_TESTING"] = False
 
-        # KB-H016: do not install Find*.cmake
-        tools.replace_path_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
-            "install(FILES ${installFindModuleFiles} DESTINATION ${FIND_MODULES_INSTALL_DIR})", "")
+        if self.package_folder is None:
+            share_folder = "res"
+        else:
+            share_folder = os.path.join(self.package_folder, "res")
 
-        self._cmake = CMake(self)
-        self._cmake.definitions["BUILD_HTML_DOCS"] = False
-        self._cmake.definitions["BUILD_QTHELP_DOCS"] = False
-        self._cmake.definitions["BUILD_MAN_DOCS"] = False
-        self._cmake.definitions["SHARE_INSTALL_DIR"] = os.path.join(self.package_folder, "res")
-        self._cmake.configure(source_folder=os.path.join(self.source_folder, self._source_subfolder))
-        return self._cmake
+        tc.cache_variables["SHARE_INSTALL_DIR"] = share_folder
+        tc.generate()
 
     def build(self):
-        cmake = self._configure_cmake()
-        cmake.build()
+        cm_folder = "{}-{}".format(self.name, self.version)
+        cmake = CMake(self)
+        cmake.configure(build_script_folder=cm_folder)
 
     def package(self):
-        cmake = self._configure_cmake()
-        cmake.install()
-        self.copy("testhelper.h", src=os.path.join(self.source_folder, self._source_subfolder, "tests/ECMAddTests"), dst="res/tests")
-        self.copy("*", src=os.path.join(self.source_folder, self._source_subfolder, "LICENSES"), dst="licenses")
+        cm_folder = "{}-{}".format(self.name, self.version)
+        lic_folder_st = os.path.join(cm_folder, "LICENSES")
+        lic_folder = os.path.join(self.source_folder, lic_folder_st)
+        self.output.info("license folder: {}".format(lic_folder))
```

Comment:
```suggestion
```

Not needed as when running package step, all copied files are listed, including license. Plus, there is will be an warning message from the client in case licenses folder is empty.

Reply:
ok, will remove this (wasn't my code, was left over from conan 1 era)